tests/integration/test_amax_amin.py

```diff
@@ -0,0 +1,237 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import numpy as np
+import pytest
+from legate.core import LEGATE_MAX_DIM
+
+import cunumeric as num
+
+FUNCS = ("amax", "amin")
+
+
+@pytest.mark.parametrize("keepdims", [True, False])
+@pytest.mark.parametrize("ndim", range(LEGATE_MAX_DIM + 1))
+@pytest.mark.parametrize("func_name", FUNCS)
+def test_basic(func_name, ndim, keepdims):
+    shape = (5,) * ndim
+    in_np = np.random.random(shape)
+    in_num = num.array(in_np)
+
+    func_np = getattr(np, func_name)
+    func_num = getattr(num, func_name)
+
+    assert np.array_equal(
+        func_np(in_np, keepdims=keepdims),
+        func_num(in_num, keepdims=keepdims),
+    )
+
+    for val in (-2, 0.5, 2):
+        assert np.array_equal(
+            func_np(in_np, keepdims=keepdims, initial=val),
+            func_num(in_num, keepdims=keepdims, initial=val),
+        )
+
+
+@pytest.mark.parametrize("keepdims", [True, False])
+@pytest.mark.parametrize("ndim", range(LEGATE_MAX_DIM + 1))
+@pytest.mark.parametrize("func_name", FUNCS)
+def test_axis(func_name, ndim, keepdims):
+    shape = (5,) * ndim
+    in_np = np.random.random(shape)
+    in_num = num.array(in_np)
+
+    func_np = getattr(np, func_name)
+    func_num = getattr(num, func_name)
+
+    axis_list = list(range(in_num.ndim))
+    axis_list.append(-ndim)
+
+    for axis in axis_list:
+        res_np = func_np(in_np, axis=axis, keepdims=keepdims)
+        res_num = func_num(in_num, axis=axis, keepdims=keepdims)
+        assert np.array_equal(res_np, res_num)
+
+        for val in (-2, 0.5, 2):
+            assert np.array_equal(
+                func_np(in_np, keepdims=keepdims, initial=val),
+                func_num(in_num, keepdims=keepdims, initial=val),
+            )
+
+
+@pytest.mark.xfail
+@pytest.mark.parametrize("axes", ((-3, -1), (-1, 0), (-2, 2), (0, 2)))
+@pytest.mark.parametrize("keepdims", [True, False])
+@pytest.mark.parametrize("func_name", FUNCS)
+def test_axis_tuple(func_name, keepdims, axes):
+    # In Numpy, it pass
+    # In cuNumeric, it raises NotImplementedError
+    shape = (3, 4, 5)
+    in_np = np.random.random(shape)
+    in_num = num.array(in_np)
+
+    func_np = getattr(np, func_name)
+    func_num = getattr(num, func_name)
+
+    res_np = func_np(in_np, axis=axes, keepdims=keepdims)
+    res_num = func_num(in_num, axis=axes, keepdims=keepdims)
+    assert np.array_equal(res_np, res_num)
+
+
+@pytest.mark.parametrize("keepdims", [True, False])
+@pytest.mark.parametrize("func_name", FUNCS)
+def test_out_dim0(func_name, keepdims):
+    shape = (5,) * 0
+    in_np = np.random.random(shape)
+    in_num = num.array(in_np)
+
+    func_np = getattr(np, func_name)
+    func_num = getattr(num, func_name)
+
+    res_np = np.random.randint(1, 10, size=())
+    res_num = num.random.randint(1, 10, size=())
```

Comment:
why `randint`? Is there an problem with `empty` or `zeros`?

Reply:
Fixed. Thanks.